snuba/datasets/processors/search_issues_processor.py

```
@@ -0,0 +1,105 @@
+    search_title: str
+    detection_timestamp: float
+
+    data: SearchIssueData
+
+
+class SearchIssuesMessageProcessor(DatasetMessageProcessor):
+    def process_message(
+        self, message: Tuple[int, str, SearchIssueEvent], metadata: KafkaMessageMetadata
+    ) -> Optional[ProcessedMessage]:
+        if not (isinstance(message, (list, tuple)) and len(message) >= 2):
+            return None
+
+        # TODO: validate version
+        version = message[0]
+        if not version:
+            return None
+
+        type_, event = message[1:3]
+        if type_ != "insert":
+            return None
+
+        try:
+            detection_timestamp = datetime.utcfromtimestamp(
+                event["detection_timestamp"]
+            )
+
+            try:
+                retention_days = enforce_retention(
+                    event.get("retention_days", 90), detection_timestamp
+                )
+            except EventTooOld:
+                metrics.increment("event_too_old")
+                return None
+
+            group_ids = event["group_ids"]
+            # TODO: make use of event.data
+            # event_data = event.get("data", {})
+
+            processed = [
+                {
+                    "project_id": event["project_id"],
+                    "organization_id": event["organization_id"],
+                    "group_id": group_id,
+                    "detection_timestamp": detection_timestamp,
+                    "retention_days": retention_days,
+                }
+                for group_id in group_ids or ()
+            ]
+        except IndexError:
+            metrics.increment("invalid_message")
+            return None
+        except ValueError:
+            metrics.increment("invalid_uuid")
+            return None
+        except KeyError:
+            metrics.increment("missing_field")
+            return None
+        except Exception:
+            metrics.increment("process_message_error")
+            raise
```

While you are initially testing this dataset, I would have nothing actually raise here. If this raises, the entire consumer will shut down and stop processing (since it will be stuck on the bad message).

---

Please let's not silently drop messages, The storage has a Dead Letter Queue configured, raising exceptions is ok.